### PR TITLE
#1097 Option to hide internal tables in Hasura

### LINKED ISSUE #1097: Option to hide internal tables in Hasura
Configure Hasura to exclude internal tables from generating a GraphQL API based on a specific naming pattern.
Indexer config should have an option to skip `dipdup_*` tables. And maybe arbitrary ones. Like this:

```yaml
hasura:
  url: ...
  ignore_internal: true
  ignore:
    - some_user_table
```

### PR DESCRIPTION
**Pull request by tosoham**
Issue #1097

Did changes in Class HasuraConfig and dipdup.hasura module to mentioned imporovements

### CODE REVIEW

File: src/dipdup/hasura.py [hunk 1]
```diff
@@ -356,18 +356,33 @@ async def _generate_source_tables_metadata(self) -> list[dict[str, Any]]:
...
             metadata_tables[view] = self._format_table(view)
 
         for app, model in iter_models(self._package):
             table_name = model_tables[f'{app}.{model.__name__}']
+            if not table_name:
+                continue
 
             for field in model._meta.fields_map.values():
                 if isinstance(field, fields.relational.ForeignKeyFieldInstance):
                     related_table_name = model_tables[field.model_name]
```
Comment (droserasprout) [inline src/dipdup/hasura.py]: This call will fail if table/view is ignored; use `get()`

File: src/dipdup/hasura.py [hunk 2]
```diff
@@ -386,8 +401,15 @@ async def _generate_source_tables_metadata(self) -> list[dict[str, Any]]:
 
                 elif isinstance(field, fields.relational.ManyToManyFieldInstance):
                     related_table_name = model_tables[field.model_name]
```
Comment (droserasprout) [inline src/dipdup/hasura.py]: This call will fail if table/view is ignored; use `get()`

File: src/dipdup/config/__init__.py [hunk 3]
```diff
@@ -375,6 +375,8 @@ class HasuraConfig:
     :param camel_case: Whether to use camelCase instead of default pascal_case for the field names.
     :param rest: Enable REST API both for autogenerated and custom queries.
     :param http: HTTP connection tunables
+    :param ignore_internal: Whether to ignore internal tables (prefixed with 'dipdup_') when generating the GraphQL API.
```
Comment (droserasprout) [inline src/dipdup/config/__init__.py:378]: Please, add an entry to `CHANGELOG.md`, then make sure that `make all docs` command passes.

**Review by droserasprout — CHANGES_REQUESTED** [review]

File: src/dipdup/hasura.py [hunk 4]
```diff
@@ -356,18 +356,33 @@ async def _generate_source_tables_metadata(self) -> list[dict[str, Any]]:
...
         for view in views:
+            if self._hasura_config.ignore_internal and view.startswith('dipdup_'):
+                continue
+            if view in self._hasura_config.ignore:
+                continue
+
             metadata_tables[view] = self._format_table(view)
 
         for app, model in iter_models(self._package):
             table_name = model_tables[f'{app}.{model.__name__}']
```
Comment (droserasprout) [inline src/dipdup/hasura.py]: This call will fail if table/view is ignored; use `get()`

**Review by tosoham — COMMENTED** [review]
I've access the model_tables dictionary with get() method

**Comment by tosoham** [comment]
Updated the CHANGELOG.md

File: CHANGELOG.md [hunk 5]
```diff
@@ -5,6 +5,13 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog], and this project adheres to [Semantic Versioning].
 
 Releases prior to 7.0 has been removed from this file to declutter search results; see the [archived copy](https://github.com/dipdup-io/dipdup/blob/8.0.0b5/CHANGELOG.md) for the full list.
+## [8.0.0] - 2024-09-28
```
Comment (droserasprout) [inline CHANGELOG.md]: Please, follow the current document format (Keep a Changelog). Example for unreleased section: https://github.com/dipdup-io/dipdup/pull/1102/files

**Review by droserasprout — CHANGES_REQUESTED** [review]

File: src/dipdup/config/__init__.py (hunk 3 above)
Comment (tosoham) [reply src/dipdup/config/__init__.py:378]: yes I've added an entry in the `CHANGELOG.md` and made sure `make all docs` command passes successfully

File: CHANGELOG.md (hunk 5 above)
Comment (tosoham) [reply CHANGELOG.md]: done!

**Review by droserasprout — APPROVED** [review]

**Pull request merged**